plugin/restore/mysql/mysql.go

```diff
@@ -88,14 +147,79 @@ func (r *Restore) RestorePITR(ctx context.Context, fullBackup *bufio.Scanner, bi
 		return err
 	}
 
-	if err := replayBinlog(ctx, binlog, r.binlogDir); err != nil {
+	if err := r.replayBinlog(ctx, database, pitrDatabaseName, r.binlogDir, binlog, suffixTs); err != nil {
 		// TODO(dragonly)/TODO(zp): Handle the error when implement replayBinlog.
 		return nil
 	}
 
 	return nil
 }
 
+func (r *Restore) getReplayBinlogPathList(ctx context.Context, startBinlogInfo api.BinlogInfo, binlogDir string) ([]string, error) {
+	binlogNamePrefix, err := r.getBinlogNamePrefix(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("cannot get the prefix of binlog name, error: %w", err)
+	}
+	return getReplayBinlogPathListImpl(startBinlogInfo, binlogDir, binlogNamePrefix)
+}
+
+func getReplayBinlogPathListImpl(startBinlogInfo api.BinlogInfo, binlogDir, binlogNamePrefix string) ([]string, error) {
+	startBinlogSeq, err := getBinlogFileNameSeqNumber(startBinlogInfo.FileName, binlogNamePrefix)
+	if err != nil {
+		return nil, fmt.Errorf("cannot parse the start binlog name [%s], error: %w", startBinlogInfo.FileName, err)
+	}
+
+	binlogFilesLocal, err := ioutil.ReadDir(binlogDir)
+	if err != nil {
+		return nil, fmt.Errorf("cannot read directory %s, error %w", binlogDir, err)
+	}
+
+	type binlogItem struct {
+		name string
+		seq  int
+	}
+	var needReplayBinlogName []binlogItem
+
+	for _, f := range binlogFilesLocal {
+		if f.IsDir() || !strings.HasPrefix(f.Name(), binlogNamePrefix) {
+			continue
+		}
+		// for mysql binlog, after the serial number reaches 999999, the next serial number will not return to 000000, but 1000000,
+		// so we cannot directly use string to compare lexicographical order.
+		binlogSeq, err := getBinlogFileNameSeqNumber(f.Name(), binlogNamePrefix)
+		if err != nil {
+			return nil, fmt.Errorf("cannot parse the start binlog name [%s], error: %w", f.Name(), err)
+		}
+		if binlogSeq >= startBinlogSeq {
+			needReplayBinlogName = append(needReplayBinlogName, binlogItem{
+				name: f.Name(),
+				seq:  int(binlogSeq),
+			})
+		}
+	}
+
+	if len(needReplayBinlogName) == 0 {
+		return []string{}, nil
+	}
+
+	sort.Slice(needReplayBinlogName, func(i, j int) bool {
+		return needReplayBinlogName[i].seq < needReplayBinlogName[j].seq
+	})
+
+	for i := 1; i < len(needReplayBinlogName); i++ {
+		if needReplayBinlogName[i].seq != needReplayBinlogName[i-1].seq+1 {
+			return nil, fmt.Errorf("detect a discontinuous binlog sequence %v", needReplayBinlogName)
```

Comment:
```suggestion
			return nil, fmt.Errorf("detect a discontinuous binlog file extension %s", needReplayBinlogName)
```

Reply:
may be kept %v to print the whole slice so that both users and we can clearly see where the gap is.

Reply:
IMO returning the wrong point is enough. If the binlog file list is long, the user will easily lose the point.

Reply:
SG. How about we point out where the gap is?

Reply:
Not quite necessary for now.